buildSrc/src/main/java/org/zaproxy/zap/tasks/HandleWeeklyRelease.java

```diff
@@ -0,0 +1,111 @@
+ * limitations under the License.
+ */
+package org.zaproxy.zap.tasks;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+
+/**
+ * Task that handles a weekly release.
+ *
+ * <p>Sends a repository dispatch to update the marketplace.
+ */
+public abstract class HandleWeeklyRelease extends SendRepositoryDispatch {
+
+    private static final String HTTPS_SCHEME = "HTTPS";
+
+    private Map<String, String> payloadData;
+
+    public HandleWeeklyRelease() {
+        getClientPayload()
+                .set(
+                        getProject()
+                                .provider(
+                                        () -> {
+                                            if (payloadData == null) {
+                                                createPayloadData();
+                                            }
+                                            return payloadData;
+                                        }));
+    }
+
+    @Input
+    public abstract Property<String> getDownloadUrl();
+
+    @Input
+    public abstract Property<String> getChecksumAlgorithm();
+
+    private void createPayloadData() {
+        String checksum;
+        try {
+            checksum = createChecksum(getChecksumAlgorithm().get(), downloadRelease());
+        } catch (IOException e) {
+            throw new BuildException(e);
+        }
+
+        payloadData = new HashMap<>();
+        payloadData.put("url", getDownloadUrl().get());
+        payloadData.put("checksum", checksum);
+    }
+
+    private Path downloadRelease() throws IOException {
+        String urlString = getDownloadUrl().get();
+        URL url = new URL(urlString);
+        if (!HTTPS_SCHEME.equalsIgnoreCase(url.getProtocol())) {
+            throw new IllegalArgumentException(
+                    "The provided URL does not use HTTPS scheme: " + url.getProtocol());
+        }
+
+        Path release = getTemporaryDir().toPath().resolve(extractFileName(urlString));
+        if (Files.exists(release)) {
+            getLogger().info("File already exists, skipping download.");
+            return release;
+        }
+
+        try (InputStream in = url.openStream()) {
+            Files.copy(in, release);
+        } catch (IOException e) {
+            throw new IOException("Failed to download the file: " + e.getMessage(), e);
+        }
+        getLogger().info("File downloaded to: " + release);
+        return release;
+    }
+
+    private static String extractFileName(String url) {
+        int idx = url.lastIndexOf("/");
+        if (idx == -1) {
+            throw new IllegalArgumentException(
+                    "The provided URL does not have a file name: " + url);
+        }
```

Comment:
Wouldn't this also be true if the index was the end (length) of the string? (https://example.org/)

Reply:
Yes, this is just a quick check that something is there (e.g. it wouldn't work properly with an empty path either).

Reply:
okay